autoload/health/nvim_lsp.vim

```diff
@@ -0,0 +1,20 @@
+
+function! health#nvim_lsp#check()
+	call health#report_start('Checking language server protocol configuration')
+lua << EOF
+	local configs = require 'nvim_lsp/configs'
+	local lsp = require 'vim.lsp'
+
+	for name, top_level_config in pairs(configs) do
+		-- the folder needs to exist, => use stdpath ?
+		local new_config = top_level_config.gen_config("/tmp")
```

Comment:
at first I had used "/fake/path" but the validator checks the folder exists so I was thinking of using `stdpath('cache')` to avoid creating a temporary directory just for checkhealth.  Question is does `stdpath('cache')` always exist ?

Reply:
> Question is does stdpath('cache') always exist ?

At least it didn't exist in my environment.
If `stdpath('config')`, init.vim will be placed, so it may not exist in many environments.

@justinmk Is there any good workaround?

Reply:
> Question is does `stdpath('cache')` always exist ?

Would need to create it here. Is there something wrong with that?

Nvim could add e.g. `ensurecache()`, but that's not different than `mkdir(stdpath('cache'), 'p')`.  Though I suppose Nvim could do this once at startup, then plugins can reasonably assume it exists...

Reply:
I've been using `.`

Reply:
isn't it better to use cache or data dirs for their intended purpose?